packages/electron-updater/src/MacUpdater.ts

```diff
@@ -50,13 +50,18 @@ export class MacUpdater extends AppUpdater {
       log.warn(`sysctl shell command to check for macOS Rosetta environment failed: ${e}`)
     }
 
+    const isArm64Mac = process.arch === "arm64" || isRosetta
+
     // allow arm64 macs to install universal or rosetta2(x64) - https://github.com/electron-userland/electron-builder/pull/5524
     const isArm64 = (file: ResolvedUpdateFileInfo) => file.url.pathname.includes("arm64") || file.info.url?.includes("arm64")
-    if (files.some(isArm64)) {
-      files = files.filter(file => (process.arch === "arm64" || isRosetta) === isArm64(file))
```

Comment:
Would you mind explaining the change you made here, @johnnyopao? From my understanding, `files = files.filter(file => (process.arch === "arm64" || isRosetta) === isArm64(file))` already handles both Intel & ARM64 cases. It would be equivalent to `files = files.filter(file => true === isArm64(file))` on ARM64 Macs and `files = files.filter(file => false === isArm64(file))` on Intel Macs.

Reply:
Ah I see, one of my friend pointed out `isRosetta === undefined` causes the problem previously.

Reply:
Ah I see, `isRosetta` could be `undefined`, causing the problem.